contracts/scripts/_discoveryprovider-test-init.js

```diff
@@ -50,10 +50,21 @@ const printUsageAndThrow = () => {
 }
 
 /** Copies the contents of build/contracts to the outputPath */
-const copyBuildDirectory = (outputPath) => {
+const copyBuildDirectory = async (outputPath) => {
   let dir = path.join(__dirname, '..')
   let localTarget = path.join(dir, 'build/contracts')
-  fs.mkdirSync(outputPath, {recursive: true})
+
+  /* Creates build/contrats folder if folder is not present */
+  async function createContractsDir (dir) {
```

Comment:
@vicky-g just a question. why have you declared this async function inside another function?

Reply:
i thought it'd be a good idea to use async for `createContractsDir` so that directory is created before doing any operations on that path

Reply:
@vicky-g similar to @hareeshnagaraj's question, what's the reasoning behind putting this inside the copyBuildDirectory function as opposed to keeping this function declaration as a standalone function?

Reply:
ah, good idea! will take it out as a util function that can be used for `build/contracts` and `libs/eth-contracts`